packages/next-tinacms-github/src/get-github-static-props.ts

```diff
@@ -0,0 +1,70 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+import { GithubError } from './github/content/GithubError'
+import { getMarkdownFile } from './github/content/getMarkdownFile'
+import { getJsonFile } from './github/content/getJsonFile'
+
+export interface PreviewMeta {
+  github_access_token: string
+  fork_full_name: string
+  head_branch: string
+  file: {
+    relativePath: string
+    name?: string
+    format?: 'markdown' | 'json'
+  }
+}
+
+export async function getGithubStaticProps(
```

Comment:
Might be good to type & export the return type so we can keep it consistent between using Github data vs non-preview-data

Reply:
Agreed. I'm going to remove that magic "switch the name" stuff